svf-llvm/include/SVF-LLVM/CppUtil.h

```diff
@@ -129,6 +129,39 @@ bool VCallInCtorOrDtor(const CallBase* cs);
 bool isSameThisPtrInConstructor(const Argument* thisPtr1,
                                 const Value* thisPtr2);
 
+/// Extract class name based on the c++ callee function, e.g., constructor
+Set<std::string> extractClassNameViaCppFunc(const Function *foo);
+
+/// Extract class name in template functions
+Set<std::string> extractClassNameInTemplate(const std::string &oname);
+
+/// CPP source used to infer c++ class name
+/// sources can be heap allocation or self-inference functions (constructors/destructors or template functions)
+bool isCPPSource(const Value *val);
+
+/// whether fooName matches the mangler label
+bool matchManglerLabel(const std::string &fooName, const std::string &label);
+
+/// whether foo is a cpp self-inference function
```

Comment:
What does this mean? Can be self-inferred?

Reply:
functions that we can get the class name from the function name, like constructors (e.g., A::A()), template functions (e.g., std::deque<A const*, std::allocator<A const*> >). Any suggestions on better naming?